app/models/rook.rb

```diff
@@ -1,3 +1,26 @@
 class Rook < Piece
 
+	def valid_move?(dest_x, dest_y)
+
+		#a move can not be a non-move
+    	return false if dest_x == self.x_position && dest_y == self.y_position
+
+		#no obstructions allowed
+		return false if is_obstructed?(dest_x, dest_y) == true
+
+		#only vertical or horizontal moves allowed
+		return false if (dest_x < self.x_position || dest_x > self.x_position) && (dest_y < self.y_position || dest_y > self.y_position)
+
+		#no vertical moves off the board
+		return false if dest_x < 1 || dest_x > 8
+    	
+    	#no horizontal moves off the board
+    	return false if dest_y < 1 || dest_y > 8
```

Comment:
@CodeFlowed @robgraphix1 I agree. I'm thinking maybe have the method in the piece model and call it when we call `validate_move` from the controller. Thoughts?